docs/sphinx/specification/cudaq/operations.rst

```diff
@@ -74,33 +48,7 @@ The default set of quantum intrinsic operations for the
       template <typename mod = base, typename ScalarAngle, typename... QubitArgs> 
       void ROTATION_NAME(ScalarAngle angle, QubitArgs &...args) noexcept { ... }
  
-      // General swap with control variants
-      // must take at least 2 qubits
-      template<typename... QubitArgs>
-      void swap(QubitArgs&... args) { ... }
```

Comment:
Why did you get rid of `swap()`? It's still a valid gate, right?

Also, since I can't make comments on the above lines (like line 39, 42, 45, 49, 51, and 52), I think that a) since we the number of gates is small enough, and b) the textual description was removed (lines 86-106), it might be worth spelling out the {x,y,z,h,t,s} gates for `NAME`, etc., just to provide clarity on this code block. Same for `ROTATION_NAME` and `MEASURE_OP`. (It actually took me a little while to figure out what this code block was trying to communicate.)

Reply:
The paragraph in the specification as it was did not document what the actual mathematical definition for each gate was. While the names are commonly recognized, there are a couple of conventions where implementations might differ (in particular for rotation gates, conventions can vary). I hence did exactly as you suggested and instead listed each gate explicitly and separately. However, convoluting that with giving the exact C++ interface, when C++ is not the format that backends ultimately consume just seemed odd, plus I would not expect the list of gates be part of a language specification (it is/should be a library of sorts). I hence split this into two pieces:
1) the actual definition of what each operation does and how to invoke it is in the API documentation.
2) the information for implementers that states what the requirements and C++ signatures are when defining your own custom gate set is listed in the language spec.

There shouldn't be a redundancy between the API documentation and the language spec (single source of truth, otherwise we just end up with self-contradicting documentation sooner or later). However, I added a sentence here that points to the API documentation and clarified the snippet a bit. Let me know if that makes it clear.